src/Cli/dotnet/Commands/Test/TestApplicationActionQueue.cs

```diff
@@ -65,18 +56,16 @@ private async Task Read(Func<TestApplication, Task<int>> action)
 
             lock (_lock)
             {
-                if (_firstExitCode == null)
+                if (_exitCode is null)
                 {
-                    _firstExitCode = result;
+                    // This is the first result we are getting.
+                    // So we assign the exit code.
+                    _exitCode = result;
```

Comment:
I like this simplification, but I'm wondering if it's correct. I think it does what the previous code did, but I'm wondering if it would be better to have success + specific failure --> specific failure rather than success + specific failure --> generic failure. Not a strong feeling.

Reply:
Good point. I think it makes sense.